src/clj/puppetlabs/services/config/puppet_server_config_core.clj

```diff
@@ -92,6 +93,18 @@
                     "conflict with the values to be "
                     "read from puppet: " key-conflicts))))))
 
+(defn check-master-url-prefix
+  "Ensure that Puppet Server's route for the puppet master matches what is in
+  puppet.conf."
+  [webrouter-config puppet-config]
+  (let [puppet-prefix (:master-url-prefix puppet-config)
+        webrouter-prefix (:puppetlabs.services.master.master-service/master-service webrouter-config)]
+    (when-not (= puppet-prefix webrouter-prefix)
+      (throw (Exception.
```

Comment:
We should pick a more specific Exception to use here, or we should consider using slingshot instead.

Basically our pattern has been to throw an exception if you intend for it to bring down the program and give a good message to the user.  On the other hand, if you intend for the "exception" to just be caught further up the stack and handled there, use slingshot.